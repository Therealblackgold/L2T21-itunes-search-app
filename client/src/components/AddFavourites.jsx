// AddFavourites renders the overlay content to add favourites
const AddFavourites = () => {
  return (
    <>
      <span style={{ marginRight: "6px" }}>Add to Favourites</span>
      <i
        className="bi bi-heart-fill"
        style={{ color: "red", marginTop: "2px" }}
      ></i>
    </>
  );
};

export default AddFavourites;
