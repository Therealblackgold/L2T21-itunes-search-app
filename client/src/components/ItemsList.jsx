// ItemList component renders a list of items returned from the api.
const ItemsList = ({ items, FavouriteComponent, handleFavouritesClick }) => {
  return (
    <>
      {items.map((item, index) => (
        <div
          key={index}
          className="col card-container  m-3"
          data-testid="listItem"
        >
          <div className="card">
            <img src={item.artworkUrl100} alt="item" />
            <h5>
              {item.artistName}, {item.trackName}
            </h5>
            <p>
              Date:{item.releaseDate}, Categories:{item.kind}{" "}
              {item.primaryGenreName}
            </p>
            <div
              onClick={() => handleFavouritesClick(item)}
              className="overlay d-flex align-items-center justify-content-center"
            >
              <FavouriteComponent />
            </div>
          </div>
        </div>
      ))}
    </>
  );
};

export default ItemsList;
