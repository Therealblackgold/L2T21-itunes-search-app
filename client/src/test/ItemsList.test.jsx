import ItemsList from "../components/ItemsList";
import AddFavourites from "../components/AddFavourites";
import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";

const mockedAddFavouriteItem = jest.fn();
const mockedItem = [
  {
    artworkUrl100:
      "https://is3-ssl.mzstatic.com/image/thumb/Music115/v4/fb/64/0a/fb640a31-440e-6855-7bf9-8368648bb74d/14UMGIM60062.rgb.jpg/100x100bb.jpg",
    artistName: "James Bay",
    trackName: "Let It Go",
    releaseDate: "05:03:2014",
    kind: "song",
    primaryGenreName: "Alternative",
  },
];

describe("ItemList Component", () => {
  test("Should render properly", () => {
    const component = renderer.create(
      <ItemsList
        items={mockedItem}
        handleFavouritesClick={mockedAddFavouriteItem}
        FavouriteComponent={AddFavourites}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Should render a list item and display item data", () => {
    render(
      <ItemsList
        items={mockedItem}
        handleFavouritesClick={mockedAddFavouriteItem}
        FavouriteComponent={AddFavourites}
      />
    );
    const listItem = screen.getByTestId("listItem");
    expect(listItem).toBeInTheDocument();
  });

  test("Should be able to render addFavourite component", () => {
    render(
      <ItemsList
        items={mockedItem}
        handleFavouritesClick={mockedAddFavouriteItem}
        FavouriteComponent={AddFavourites}
      />
    );
    const addFavourite = screen.getByText(/add to Favourites/i);
    expect(addFavourite).toBeInTheDocument();
  });

  test("Should call addFavouriteItem function when clicked", () => {
    render(
      <ItemsList
        items={mockedItem}
        handleFavouritesClick={mockedAddFavouriteItem}
        FavouriteComponent={AddFavourites}
      />
    );
    const addFavouriteButton = screen.getByText(/add to favourites/i);
    fireEvent.click(addFavouriteButton);
    expect(mockedAddFavouriteItem).toHaveBeenCalledTimes(1);
  });
});
