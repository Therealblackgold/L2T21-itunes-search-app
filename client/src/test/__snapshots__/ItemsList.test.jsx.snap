// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList Component Should render properly 1`] = `
<div
  className="col card-container  m-3"
  data-testid="listItem"
>
  <div
    className="card"
  >
    <img
      alt="item"
      src="https://is3-ssl.mzstatic.com/image/thumb/Music115/v4/fb/64/0a/fb640a31-440e-6855-7bf9-8368648bb74d/14UMGIM60062.rgb.jpg/100x100bb.jpg"
    />
    <h5>
      James Bay
      , 
      Let It Go
    </h5>
    <p>
      Date:
      05:03:2014
      , Categories:
      song
       
      Alternative
    </p>
    <div
      className="overlay d-flex align-items-center justify-content-center"
      onClick={[Function]}
    >
      <span
        style={
          Object {
            "marginRight": "6px",
          }
        }
      >
        Add to Favourites
      </span>
      <i
        className="bi bi-heart-fill"
        style={
          Object {
            "color": "red",
            "marginTop": "2px",
          }
        }
      />
    </div>
  </div>
</div>
`;-
